### PR TITLE
[SPARK-48301][SQL] Rename `CREATE_FUNC_WITH_IF_NOT_EXISTS_AND_REPLACE` to `CREATE_ROUTINE_WITH_IF_NOT_EXISTS_AND_REPLACE`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Rename `CREATE_FUNC_WITH_IF_NOT_EXISTS_AND_REPLACE` to `CREATE_ROUTINE_WITH_IF_NOT_EXISTS_AND_REPLACE`

### Why are the changes needed?
`IF NOT EXISTS` + `REPLACE` is standard restriction, not just for functions.
Rename it to make it reusable.

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
updated tests

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @srielau and @allisonwang-db

**Review by HyukjinKwon — APPROVED** [review]

File: common/utils/src/main/resources/error/error-conditions.json [hunk 1]
```diff
@@ -2675,9 +2675,9 @@
           "ANALYZE TABLE(S) ... COMPUTE STATISTICS ... <ctx> must be either NOSCAN or empty."
         ]
       },
-      "CREATE_FUNC_WITH_IF_NOT_EXISTS_AND_REPLACE" : {
+      "CREATE_ROUTINE_WITH_IF_NOT_EXISTS_AND_REPLACE" : {
         "message" : [
-          "CREATE FUNCTION with both IF NOT EXISTS and REPLACE is not allowed."
+          "CREATE ROUTINE with both IF NOT EXISTS and REPLACE is not allowed."
```
Comment (srielau) [inline common/utils/src/main/resources/error/error-conditions.json:2680]:
```suggestion
          "CREATE PROCEDURE or CREATE FUNCTION with both IF NOT EXISTS and REPLACE is not allowed."
```

Comment (srielau) [reply common/utils/src/main/resources/error/error-conditions.json:2680]: routine is a concept, like "relation". It's not a keyword/object

**Review by srielau — APPROVED** [review]

**Comment by zhengruifeng** [comment]
merged to master

File: common/utils/src/main/resources/error/error-conditions.json [hunk 2]
```diff
@@ -2675,9 +2675,9 @@
           "ANALYZE TABLE(S) ... COMPUTE STATISTICS ... <ctx> must be either NOSCAN or empty."
         ]
       },
-      "CREATE_FUNC_WITH_IF_NOT_EXISTS_AND_REPLACE" : {
+      "CREATE_ROUTINE_WITH_IF_NOT_EXISTS_AND_REPLACE" : {
         "message" : [
-          "CREATE FUNCTION with both IF NOT EXISTS and REPLACE is not allowed."
+          "CREATE PROCEDURE or CREATE FUNCTION with both IF NOT EXISTS and REPLACE is not allowed."
```
Comment (allisonwang-db) [inline common/utils/src/main/resources/error/error-conditions.json:2680]: How about: "Cannot create a routine with both IF NOT EXISTS and REPLACE specified."

Comment (srielau) [reply common/utils/src/main/resources/error/error-conditions.json:2680]: That works too

Comment (zhengruifeng) [reply common/utils/src/main/resources/error/error-conditions.json:2680]: sounds good, we don't have PROCEDURE in OSS, will update in a followup PR

**Pull request closed without merging**